src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -42,8 +42,13 @@ public IOperation Create(BoundNode boundNode)
             return _cache.GetOrAdd(boundNode, n => CreateInternal(n));
         }
 
-        private IOperation CreateInternal(BoundNode boundNode)
+        private IOperation CreateInternal(BoundNode boundNode, bool createParenthesized = true)
```

Comment:
>bool createParenthesized = true [](start = 63, length = 31)

I think there is a better alternative to adding the optional parameter, i.e.:
- rename ```CreateParenthesizedExpression``` to ```CreateInternal```, make necessary adjustments 
- rename ```CreateInternal``` to something else, call it from the new CreateInternal and from CreateUnboundLambdaOperation.